src/solve/nonconvex/nonconvex_bayesian.jl

```diff
@@ -0,0 +1,45 @@
+function convert_common_kwargs(opt::NonconvexBayesian.BayesOptAlg, opt_kwargs;
+    cb=nothing,
+    maxiters=nothing,
+    maxtime=nothing,
+    abstol=nothing,
+    reltol=nothing)
+
+    conv_opt_kwargs = (; opt_kwargs...)
+
+    if !isnothing(cb)
+        @warn "common callback argument is currently not used by $(opt)"
+    end
+  
+    if !isnothing(maxiters)
+        conv_opt_kwargs = (; conv_opt_kwargs..., maxiter=maxiters)
+    end
+
+    if !isnothing(maxtime)
+        @warn "common maxtime argument is currently not used by $(opt)"
+    end
+
+    if !isnothing(abstol)
+        @warn "common abstol argument is currently not used by $(opt)"
+    end
+    
+    if !isnothing(reltol)
+        conv_opt_kwargs = (; conv_opt_kwargs..., ftol =reltol)
```

Comment:
Different sub-solvers will use different names for the ftol kwarg. I don't attempt to unify these in Nonconvex. I don't think this line will do what's expected in general.

Reply:
There is a mapping function for each `Nonconvex` optimiser. The mapping is happening only at the first level. Suboptions need to use the `Nonconvex` naming for tolerances, iters etc.

Reply:
We need to at least try to unify, with fallbacks of course.

Reply:
A quick design question: We could make it so that each all suboptions even if nested are just defined as a named tuple and the `Options` struct for `Nonconvex` is automatically inferred. 

Additionally, we could also do the mapping of common arguments beyond the top level solver such that `sub_options=(; maxiters=1000)` always works and the user does not have to know what subsolver specific keyword is.

Lastly, should `sub_option` become `local_options` mirroring what we have for other interfaces?